src/main/java/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Transactional
+    @Override
+    public ShoppingCartResponseDto getShoppingCartById(Long id) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can`t get shopping cart by id: " + id));
+        Set<CartItemResponseDto> cartItemResponseDtos = cartItemMapper
+                .toCartItemResponseDtos(shoppingCart.getCartItems());
+        ShoppingCartResponseDto shoppingCartResponseDto = shoppingCartMapper
+                .toShoppingCartResponseDto(shoppingCart);
+        shoppingCartResponseDto.setCartItems(cartItemResponseDtos);
+        return shoppingCartResponseDto;
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartResponseDto addBookToShoppingCart(
+            Long id, AddBookToCartRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can`t get shopping cart by id: " + id));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        CartItem existingCartItem = cartItems.stream()
+                .filter(cartItem -> cartItem.getBook().getId().equals(requestDto.getBookId()))
+                .findFirst()
+                .orElse(null);
+        if (existingCartItem != null) {
+            existingCartItem.setQuantity(requestDto.getQuantity() + existingCartItem.getQuantity());
+        } else {
+            CartItem cartItem = new CartItem();
+            cartItem.setShoppingCart(shoppingCart);
+            cartItem.setQuantity(requestDto.getQuantity());
+            cartItem.setBook(bookRepository.findById(requestDto.getBookId())
+                    .orElseThrow(() -> new EntityNotFoundException("Can`t get book by id: "
+                            + requestDto.getBookId())));
+            cartItem = cartItemRepository.save(cartItem);
+            cartItems.add(cartItem);
+        }
+        shoppingCart.setCartItems(cartItems);
+        shoppingCart = shoppingCartRepository.save(shoppingCart);
+        ShoppingCartResponseDto shoppingCartResponseDto = shoppingCartMapper
+                .toShoppingCartResponseDto(shoppingCart);
+        shoppingCartResponseDto.setCartItems(cartItemMapper.toCartItemResponseDtos(cartItems));
+        return shoppingCartResponseDto;
+    }
+
+    @Override
+    public CartItemResponseDto updateQuantityBook(Long cartItemId,
+                                                  UpdateQuantityBookRequestDto requestDto) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
```

Comment:
Example of implementation
find shop cart
find cart Item `findByIdAndShoppingCartId`
change quantity
save cartItem